nixos/modules/services/networking/oink.nix

```diff
@@ -20,14 +20,38 @@ in
   options.services.oink = {
     enable = lib.mkEnableOption "Oink, a dynamic DNS client for Porkbun";
     package = lib.mkPackageOption pkgs "oink" { };
+    apiKeyFile = lib.mkOption {
+      type = lib.types.nullOr lib.types.path;
+      default = null;
+      example = "/run/keys/oink-api-key";
+      description = "Path to a file containing the API key to use when modifying DNS records.";
+    };
+    secretApiKeyFile = lib.mkOption {
+      type = lib.types.nullOr lib.types.path;
+      default = null;
+      example = "/run/keys/oink-secret-api-key";
+      description = "Path to a file containing the secret API key to use when modifying DNS records.";
+    };
     settings = {
       apiKey = lib.mkOption {
         type = lib.types.str;
-        description = "API key to use when modifying DNS records.";
+        default = "";
+        example = "1234567890";
+        description = ''
+          API key to use when modifying DNS records.
+
+          This option is deprecated. Use {option}`apiKeyFile` instead.
```

Comment:
I don't think this should be deprecated right now, just discouraged.

Same thing below for secretApiKey